go.mod

```diff
@@ -49,6 +49,7 @@ require (
 	golang.org/x/sync v0.6.0
 	google.golang.org/grpc v1.61.0
 	google.golang.org/protobuf v1.32.0
+	resenje.org/singleflight v0.4.1
```

Comment:
Double checking here, I think you mean to use the x/sync version of SF.

Reply:
Actually this was very purposeful. See https://github.com/openfga/openfga/pull/1335#discussion_r1488920233